config/localTemplate.go

```diff
@@ -78,6 +78,31 @@ type Local struct {
 	CadaverSizeTarget uint64 `version[0]:"1073741824" version[24]:"0"`
 	CadaverDirectory  string `version[27]:""`
 
+	// Data directories
+	// HotDataDir is an alternative to DataDir for storing data that is frequently accessed
+	// if not set, resources which use HotDataDir should default to DataDir
+	HotDataDir string `version[0]:""`
+	// ColdDataDir is an alternative to DataDir for storing data that is infrequently accessed
+	// if not set, resources which use ColdDataDir should default to DataDir
+	ColdDataDir string `version[0]:""`
+	// TrackingDbFilePath is the path to the db file (sqlite) or directory (kv stores) used by the trackerdb
+	// by default, this is ledger.tracker.sqlite in your Genesis Directory
```

Comment:
Presumably this also defaults to HotDataDir / DataDir? Maybe worth calling out directly in the description. In the future these comments will hopefully be automatically scraped for the developer portal documentation.

Reply:
I'm now seeing you mention `Genesis Directory`. I guess this is the same thing as `DataDir`?

Reply:
Yep, my intention is to do a pass on all the comments once things settle down, but I welcome //UX comments in the meantime :)

Reply:
Sorry, I wrote my comment without refreshing to see your update. To answer your question -- Genesis Directory is a thing that is made by the Node, it's the `myNetwork-v1` folder in your dataDir. This directory is handed down and used by components like the ledger, who use it to make `ledgerDBPrefix`. I believe it's done this way to ensure that resources generated by a genesis are isolated to one another.

Hot and Cold DataDir get their own HotGenesisDir and ColdGenesisDir, so that the same fundamental isolation still exists on in those places.

Reply:
Gotcha. Yes I'm familiar with that directory but hadn't heard it called that before. Thanks for the details.

Reply:
You made me curious, since I also hadn't heard this term be used so concretely before --

In our codebase presently there are *9* places we make a variable called `genesisDir`, and 8/9 of those it's a join of `dataDir` and `genesis.ID`. Many of these are inside of other compiled `cmd`s. The standout is *[just the genesis ID* used in NodeConfigurator](https://github.com/algorand/go-algorand/blob/3b197f48b185741c9e88cfa5f3df39fd4c123c93/netdeploy/remote/nodecfg/nodeConfigurator.go#L113).

There's also a less common use of `keyDir := filepath.Join(dataDir, genID)`, and I've seen at least one use of `outdir` and `genesisFolder` in my searches. To make matters more confusing, `genesisFolder` does not have an agreed upon meaning it seems. In [deployed network, it's equal to root/genesisdata](https://github.com/algorand/go-algorand/blob/3b197f48b185741c9e88cfa5f3df39fd4c123c93/netdeploy/remote/deployedNetwork.go#L326), while in [nodeControl, it uses the genesisID like the genesisDirs](https://github.com/algorand/go-algorand/blob/3b197f48b185741c9e88cfa5f3df39fd4c123c93/nodecontrol/algodControl.go#L338)